cartridges/openshift-origin-cartridge-haproxy/bin/control

```diff
@@ -175,17 +179,22 @@ function _reload_service() {
 
 function start() {
     _start_haproxy_service
-    isrunning  &&  echo "HAProxy instance is started"
+    isrunning  &&  client_result "HAProxy instance is started"
 }
 
 function stop() {
     _stop_haproxy_service
-    isrunning  ||  echo "HAProxy instance is stopped"
+    if `pgrep -x haproxy > /dev/null 2>&1`; then
+      client_error "Could not stop haproxy, use restart or application force-stop to kill."
+      exit 1
+    else
+      isrunning  || client_result "HAProxy instance is stopped"
```

Comment:
`isrunning` returns 1 if the haproxy pid file doesn't exist. If the pid file doesn't exist (due to some race condition) and haproxy is still running, `isrunning` will still return 1. It isn't an accurate check. Perhaps `isrunning` should be modified.